### PR TITLE
br: support access by azure sas token

### PR DESCRIPTION
**Pull request by Leavrth**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

support access blob by [SAS (shared access signatures)](https://learn.microsoft.com/en-us/rest/api/storageservices/delegate-access-with-shared-access-signature).
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/14234
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/backup-and-restore-storages.md [hunk 1]
```diff
@@ -76,7 +76,8 @@ This section describes the URI format of the storage services:
 
     - `account-name`: Specifies the account name of the storage.
     - `account-key`: Specifies the access key.
-    - `access-tier`: Specifies the access tier of the uploaded objects, for example, `Hot`, `Cool`, or `Archive`. The value is `Hot` by default.
+    - `sas-token`: Specifies the shared access signature.
+    - `access-tier`: Specifies the access tier of the uploaded objects, for example, `Hot`, `Cool`, or `Archive`. The default value is the container's default access tier.
```
Comment (Oreoxmt) [inline br/backup-and-restore-storages.md:80]:
```suggestion
    - `access-tier`: Specifies the access tier of the uploaded objects, for example, `Hot`, `Cool`, or `Archive`. The default value is the default access tier of the storage account.
```

File: br/backup-and-restore-storages.md [hunk 2]
```diff
@@ -185,11 +186,15 @@ You can configure the account used to access GCS by specifying the access key. I
 </div>
 <div label="Azure Blob Storage" value="azure">
 
-- Method 1: Specify the access key
+- Method 1: Specify the shared access signature
+
+    If you specify `account-name` and `sas-token` in the URI, the authentication is performed using the specified account name and sas token. Please notice that the sas token contains the `&` character, you need encode it as `%26` before add into the URI. You can also directly encodes the entire sas token.
```
Comment (Oreoxmt) [inline br/backup-and-restore-storages.md:191]:
```suggestion
    If you specify `account-name` and `sas-token` in the URI, the authentication is performed using the specified account name and shared access signature (SAS) token. Note that the SAS token contains the `&` character, you need to encode it as `%26` before appending it to the URI. You can also directly encode the entire `sas-token` using percent-encoding.
```

**Review by Oreoxmt — APPROVED** [review]
/lgtm

**Review by YuJuncen — APPROVED** [review]
Maybe replace `percent-encoding` with `URI component encoding`.

File: br/backup-and-restore-storages.md [hunk 3]
```diff
@@ -185,11 +186,15 @@ You can configure the account used to access GCS by specifying the access key. I
 </div>
 <div label="Azure Blob Storage" value="azure">
 
-- Method 1: Specify the access key
+- Method 1: Specify the shared access signature
+
+    If you specify `account-name` and `sas-token` in the URI, the authentication is performed using the specified account name and shared access signature (SAS) token. Note that the SAS token contains the `&` character, you need to encode it as `%26` before appending it to the URI. You can also directly encode the entire `sas-token` using percent-encoding.
```
Comment (qiancai) [inline br/backup-and-restore-storages.md:191]:
```suggestion
    If you specify `account-name` and `sas-token` in the URI, the authentication is performed using the specified account name and shared access signature (SAS) token. Note that the SAS token contains the `&` character. You need to encode it as `%26` before appending it to the URI. You can also directly encode the entire `sas-token` using percent-encoding.
```

File: br/backup-and-restore-storages.md [hunk 4]
```diff
@@ -76,7 +76,8 @@ This section describes the URI format of the storage services:
 
     - `account-name`: Specifies the account name of the storage.
     - `account-key`: Specifies the access key.
-    - `access-tier`: Specifies the access tier of the uploaded objects, for example, `Hot`, `Cool`, or `Archive`. The value is `Hot` by default.
+    - `sas-token`: Specifies the shared access signature.
```
Comment (qiancai) [inline br/backup-and-restore-storages.md:79]:
```suggestion
    - `sas-token`: Specifies the shared access signature (SAS) token.
```

**Review by qiancai — APPROVED** [review]
/lgtm

**Pull request merged**